sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -593,6 +593,33 @@ case class Levenshtein(left: Expression, right: Expression) extends BinaryExpres
 }
 
 /**
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
+ */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def nullSafeEval(string: Any): Any = {
+    if (string.asInstanceOf[UTF8String].getBytes.length == 0) {
+      return string
+    }
+    else {
+      val sb = new StringBuffer()
+      sb.append(string)
+      sb.setCharAt(0, sb.charAt(0).toUpper)
+      for (i <- 1 until sb.length) {
+        if (sb.charAt(i - 1).equals(' ')) {
+          sb.setCharAt(i, sb.charAt(i).toUpper)
+        }
+      }
+      UTF8String.fromString(sb.toString)
```

Comment:
My idea would be that we check if the next character fits `Char`. If yes we convert it to Char, call `Character.toUpperCase(c)` and change the result int the array. If we cannot convert it to Char, we "ignore" it and don't change it. But as Reynold mentioned, we can do this in a second step.